File: client/java/src/main/java/io/be/mesh/mpc/Provider.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.mpc;

import io.be.mesh.macro.SPI;

/**
 * @author coyzeng@gmail.com
 */
@SPI("grpc")
public interface Provider extends AutoCloseable {

    /**
     * Start the mesh broker.
     *
     * @throws Throwable cause
     */
    void start() throws Throwable;

}

File: examples/python/src/provider.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#

from consumer import Consumer
from mesh import mps


@mps
class Provider(Consumer):

    def who(self) -> str:
        return "I am terminator"
